examples/webui-ext/readme.md

```diff
@@ -0,0 +1,53 @@
+# Web UI Extensions in Sidekiq 7.3 and 8.0
+
+This is an example of how to extend Sidekiq's Web UI to add your own tab.
+
+## Initialization
+
+The Ruby code for your extension goes in `lib/` like any normal Rubygem.
+When the user includes the `my-gem` gem within their bundle, Bundler will call `require "my-gem"` which corresponds to `lib/my-gem.rb` in your project.
+This file should **not** require any web code or assets because you don't know if you're running within a Sidekiq process or a Web process.
+
+> Don't require your web extensions in Sidekiq's initializer. This is a common mistake.
+
+Your web extension should be activated by the user calling `require "my-gem/web"` in Rails' `config/routes.rb` as that guarantees we are in a Web process.
+
+## Registration
+
+Your Web extension must register itself with Sidekiq::Web to appear in the Sidekiq UI.
+See `lib/sidekiq-redis_info/web.rb` for a full-featured example.
+
+## Content Security
+
+Sidekiq 7.3 and 8.0 explicitly lock down the Web UI to prevent XSS and malicious content injection.
+In Sidekiq 7.3, the Web UI no longer allows ěmbedding JavaScript directly on the HTML page via
+`<script>...code...</script>`. In Sidekiq 8.0, inline styling via `<style>...css...</style>` will not work.
+
+### JavaScript
+
+JavaScript must be packaged in predefined .js files with a per-request nonce that proves it was rendered by the webapp:
+
+```html
+<script type="text/javascript" src="<%= root_path %>my-gem/js/somefile.js" nonce="<%= csp_nonce %>"></script>
+```
+
+### CSS
+
+Static CSS must be linked with the nonce:
+
+```
+<link href="<%= root_path %>redis_info/css/somefile.css" media="screen" rel="stylesheet" type="text/css" nonce="<%= csp_nonce %>" />
```

Comment:
Perhaps now is the time to put a bit more thought into this: Styles/Scripts from the same origin don't need to supply a nonce. I added this somewhat defensivly per that one PR comment I read.

If the webui is served from `https://example.com/sidekiq` and styles/scripts are hosted under `https://example.com/sidekiq/assets` then all is fine. Is there actually a supported usecase where assets may be served from a different domain than where the webui is running from, like `https://s3.amazonaws.com/my_corp/assets/`? If that is not possible then adding nonces to these is not needed.

Reply:
We could provide a helper method to include a script or style tag:

```erb
<%= stylesheet("redis_info/css/somefile.css", attribute: "value") %> <!-- local, nonce not required -->
<%= stylesheet("https://cdn.example.com/redis_info/css/somefile.css") %><!-- global, will add nonce -->

<%= javascript("redis_info/js/somefile.js", attribute: "value") %> <!-- local, nonce not required -->
<%= javascript("https://cdn.example.com/redis_info/js/somefile.js") %><!-- global, will add nonce -->
```

Reply:
Yeah, that could be nice. Rails helpers have a `nonce` keyword option which transparently handles this like you proposed.

Reply:
Implemented.

Reply:
This looks fine, I think. The interpolation for key/value look a bit scary? Though I suppose it's always called with static parameters from the extension so I doubt it would matter much. Offhand I don't know if CGI even offers a method to escape attribute key/values.

You've left a debug puts in the html_tag method